src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx

```diff
@@ -72,7 +85,105 @@ const MeasureDefinitions = (props: MeasureDefinitionsProps) => {
     }
   }, [setMeasureDefinitions, measure]);
 
-  const handleSubmit = (values: MeasureDefinition) => {};
+  // Toast utilities
+  const [toastOpen, setToastOpen] = useState<boolean>(false);
+  const [toastMessage, setToastMessage] = useState<string>("");
+  const [toastType, setToastType] = useState<string>("danger");
+  const onToastClose = () => {
+    setToastType("danger");
+    setToastMessage("");
+    setToastOpen(false);
+  };
+  const handleToast = (type, message, open) => {
+    setToastType(type);
+    setToastMessage(message);
+    setToastOpen(open);
+  };
+  //first sort by term then by definition
+  const sortDefinitions = (
+    definitions: MeasureDefinition[]
+  ): MeasureDefinition[] => {
+    const sortFunction = (a: MeasureDefinition, b: MeasureDefinition) => {
+      const term1 = a.term ? a.term.toLowerCase() : "";
+      const term2 = b.term ? b.term.toLowerCase() : "";
+      const definition1 = a.definition.toLowerCase();
+      const definition2 = b.definition.toLowerCase();
+      if (term1 < term2) {
+        return -1;
+      }
+      if (term1 > term2) {
+        return 1;
+      }
+      if (term1 === term2) {
+        if (definition1 < definition2) {
+          return -1;
+        }
+        if (definition1 > definition2) {
+          return 1;
+        }
+      }
+      return 0;
+    };
+    return definitions.sort(sortFunction);
+  };
+
+  const handleSubmit = (values: MeasureDefinition) => {
+    const copiedMetaData = { ...measure?.measureMetaData };
+    // if metadata has measureDefinitions
+    if (
+      copiedMetaData.hasOwnProperty("measureDefinitions") &&
+      Array.isArray(copiedMetaData.measureDefinitions)
+    ) {
+      if (!selectedDefinition) {
+        // adding a new measure definition
+        copiedMetaData.measureDefinitions.push(values);
+        copiedMetaData.measureDefinitions = sortDefinitions(
+          copiedMetaData.measureDefinitions
+        );
+      } else {
+        //edit a measure defintion
+        const newDefinitions: Array<MeasureDefinition> =
+          copiedMetaData.measureDefinitions.filter(
+            (definition) => definition.id !== selectedDefinition.id
+          );
+        newDefinitions.push(values);
+        copiedMetaData.measureDefinitions = sortDefinitions(newDefinitions);
+      }
+    } else {
+      // if metadata does not have measureDefinitions
+      copiedMetaData.measureDefinitions = [values];
+    }
+    const modifiedMeasure = {
+      ...measure,
+      measureMetaData: copiedMetaData,
+    };
+    measureServiceApi
+      .updateMeasure(modifiedMeasure)
+      .then((res) => {
+        //@ts-ignore
+        const { status, data } = res;
+        if (status === 200) {
```

Comment:
if there is any status other than 200, can we provide some generic error in else?

Reply:
@adongare sure will do. Thanks!